src/types/types.js

```diff
@@ -0,0 +1,21 @@
+export type TypeLayer = {
+  id: number,
+  type: string,
+  component: string,
+  x1: number,
+  y1: number,
+  x2: number,
+  y2: number,
+  options: {}
+};
+
+export type TypeLayerPredicate = (layer: TypeLayer) => boolean;
+
+export type TypeMatrix = {
+  a: number,
+  b: number,
+  c: number,
+  d: number,
+  e: number,
+  f: number
+};
```

Comment:
Let's leave off the `Type` prefix on the identifiers. We've not got much else using PascalCase except components, so it should be pretty easy to identify types where they're used and there's unlikely to be many naming conflicts.

Reply:
I have implemented this change but because I am curious, even if naming conflicts are unlikely to occur, isn't this a small price to pay to avoid any future problems in this area?

Reply:
I think it's a bit of an arbitrary distinction to call out types as needing a special prefix. We don't prefix every class with "ClassSomething" or every component with "ComponentSomething". That's because we're used to them and our tooling (editors, linters etc) can automatically identify them for us. Yet we rarely run into naming conflicts with those. So what problem are we solving by naming types is a special way?

If it was trivial to do, then not worries, but a convention like this is also pretty hard to follow consistently. Would interfaces by prefixed with "Type"? What about classes ([they can be used as types](https://flow.org/en/docs/types/classes/))? Pretty quickly you find yourself and in world where you're trying to guess if the type you want to used in prefixed with "Type" or not. 🙁

The other thing that typically happens is that you want to name a type "SomethingType"… a hypothetical example in this project would be a type called "Component" and another called "ComponentType" (the former describes an instance of a component, the latter is the component's definition as it is in the config file). If you're dogmatically prefixing everything with "Type" you'll end up with "TypeComponentType". 🙂

Anyway, long story short: naming things is hard and conventions are hard to enforce. I like to keep it simple and deal with the edge cases as they come up. Haha.